lib/ansible/modules/files/file.py

```diff
@@ -76,6 +75,30 @@
     type: bool
     default: 'yes'
     version_added: "1.8"
+  modification_time:
+    description:
+      - This parameter indicates to which time should the file's modification time be set
+      - 'Should be C(same) (default) when no modification is required, C(YYYYMMDDHHMM.SS) when using default time format, or C(now)'
+    default: same
```

Comment:
Should the default be `now` instead as that's what is the current behaviour when `state: touch`?

Reply:
Since those parameters are used for every states available, if I change the default to ```now```, we will change the current behaviour of all states (except ```state: touch```) and loose their idempotence

Reply:
I see what you mean, maybe not having it defined (by default) keeps the same behaviour where touch will always update and the other states won't update. If a user explicitly defines, `now`, `same`, or `<timestamp>` then we can set accordingly.

Reply:
If we can get away with it, I'd like to change this to same so that touch is idempotent by default.  But probably need to run it by the other committers.  It is backwards incompatible but more sensible API

(If the other committers den backwards compatibility to be that important, then jborean's proposal sounds best.  If timestamp is None: use correct default for the state specified)

Reply:
Sure, awaiting more reviews then :)

Reply:
At yesterday's meeting we didn't quite decide the question of the default value decisively (only three people registered an opinion on it) but of the three there seemed to be more enthusiasm for adding the ability for "state=file" to create files (phased in via a deprecation period because of backwards compatibility concerns) rather than having touch default to keeping the same timestamp.  So if you want to be proactive, you should probably update this to have the split defaults as @jborean93 suggested.

There was some bikeshedding about the special names as well:
* There was a desire to make None mean "same" but I don't think we can do that with the split default (None means, do the default.  And since it has two meanings, we need a different special word to mean keep the timestamp the same)
* In the end people thought that "preserve" would be better than "same" (since that's used in copy to keep a file source's mode which is somewhat similar.)